### ISSUE TITLE
Parse alignment logs, stop pipeline if an alignment error occurs

### ISSUE THREAD

**Issue by chrisprobert** [kundajelab/bds_atac#19]
We need to parse the `.align.log` files generated from running bowtie and immediately kill the pipeline if bowtie exits with an error. Currently, the pipeline continues through all subsequent steps with a truncated alignment, which is bad because it produces misleading and incomplete results.

An example of a bowtie log file:
```
Error, fewer reads in file specified with -1 than in file specified with -2
terminate called after throwing an instance of 'int'
(ERR): bowtie2-align died with signal 6 (ABRT)
```

**Comment by chrisprobert** [comment]
Looking through some other logs, here are additional values that we need to detect and stop the pipeline for:

```
Error: Read NS500735:273:H2W7CBGXY:4:22504:10114:5463 1:N:0:AGCCAGAA has more read characters than quality values.
terminate called after throwing an instance of 'int'
(ERR): bowtie2-align died with signal 6 (ABRT)
```

```
gzip: /srv/gsfs0/projects/kundaje/users/cprobert/data/atac-projects/TBFFCRC_ATAC/fastqs_trimmed/ATACpt7P1_S10_R1.trimmed.fastq.gz: unexpected end of file
```

**Comment by leepc12** [comment]
fixed in commit 7431125 (bds_atac) and 5f6f612(chipseq)

https://github.com/kundajelab/TF_chipseq_pipeline/commit/5d6f612613a8779284b84c09c021d5ab73680239
